cornedbeef/src/main/java/com/swiftkey/cornedbeef/PunchHoleCoachMark.java

```diff
@@ -88,13 +117,70 @@ protected void updateView(CoachMarkDimens<Integer> popupDimens,  CoachMarkDimens
                 horizontalPadding, verticalPadding + lowerGap);
     }
 
+    /**
+     * Move the punch hole from left to right of the target view, unless the
+     * width of the target view is smaller than the diameter of the punch hole
+     * in which case the circle will be centered and the animation is pointless.
+     */
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
+    private void animateHorizontalTranslation() {
+        if (hasHorizontalTranslation() && mHorizontalAnimators == null) {
+            final int leftMostPosition = mTargetViewLoc[0] + (int) mRelCircleRadius;
+            final int rightMostPosition = mTargetViewLoc[0] + mTargetView.getWidth() - (int) mRelCircleRadius;
+
+            final int startX = isRtlConfig() ? rightMostPosition : leftMostPosition;
+            final int endX = isRtlConfig() ? leftMostPosition : rightMostPosition;
+
+            final ValueAnimator[] horizontalAnimations = new ValueAnimator[]{
+                    ObjectAnimator.ofInt(mPunchHoleView, "circleCenterX", startX, endX),
```

Comment:
It should be okay for all projects that include one of the default pro guard rule sets. I added a section in the readme for those who are rolling their own more restrictive rule set